packages/astro/src/content/runtime.ts

```diff
@@ -529,33 +531,44 @@ export function createGetLiveCollection({
 			if (processedEntries.length > 0) {
 				const entryTags = new Set<string>();
 				let minMaxAge: number | undefined;
+				let latestModified: Date | undefined;
 
 				for (const entry of processedEntries) {
 					if (entry.cacheHint) {
 						if (entry.cacheHint.tags) {
 							entry.cacheHint.tags.forEach((tag) => entryTags.add(tag));
 						}
 						if (typeof entry.cacheHint.maxAge === 'number') {
-							minMaxAge =
-								minMaxAge === undefined
-									? entry.cacheHint.maxAge
-									: Math.min(minMaxAge, entry.cacheHint.maxAge);
+							if (minMaxAge === undefined || entry.cacheHint.maxAge < minMaxAge) {
+								minMaxAge = entry.cacheHint.maxAge;
+							}
+						}
+						if (entry.cacheHint.lastModified instanceof Date) {
+							if (latestModified === undefined || entry.cacheHint.lastModified > latestModified) {
+								latestModified = entry.cacheHint.lastModified;
+							}
 						}
 					}
 				}
 
 				// Merge collection and entry cache hints
-				if (entryTags.size > 0 || minMaxAge !== undefined || cacheHint) {
-					const mergedCacheHint: any = {};
+				if (entryTags.size > 0 || minMaxAge !== undefined || latestModified || cacheHint) {
+					const mergedCacheHint: CacheHint = {};
 					if (cacheHint?.tags || entryTags.size > 0) {
-						mergedCacheHint.tags = [...(cacheHint?.tags || []), ...entryTags];
+						mergedCacheHint.tags = [...new Set([...(cacheHint?.tags || []), ...entryTags])];
 					}
 					if (cacheHint?.maxAge !== undefined || minMaxAge !== undefined) {
 						mergedCacheHint.maxAge =
 							cacheHint?.maxAge !== undefined && minMaxAge !== undefined
 								? Math.min(cacheHint.maxAge, minMaxAge)
 								: (cacheHint?.maxAge ?? minMaxAge);
 					}
+					if (cacheHint?.lastModified && latestModified) {
+						mergedCacheHint.lastModified =
+							cacheHint.lastModified > latestModified ? cacheHint.lastModified : latestModified;
+					} else if (cacheHint?.lastModified || latestModified) {
+						mergedCacheHint.lastModified = cacheHint?.lastModified ?? latestModified;
+					}
```

Comment:
Out of curiosity, do timezones have a role in this logic?

Reply:
No, they're Date objects so it compares the timestamp